src/main/java/net/dv8tion/jda/core/JDABuilder.java

```diff
@@ -119,52 +116,23 @@ public JDABuilder setToken(String token)
         return this;
     }
 
-    /**
-     * Sets the proxy that will be used by <b>ALL</b> JDA instances.
-     * <br>Once this is set <b>IT CANNOT BE CHANGED.</b>
-     * <br>After a JDA instance as been created, this method can never be called again, even if you are creating a new JDA object.
-     * <br><b>Note:</b> currently this only supports HTTP proxies.
-     *
-     * @param  proxy
-     *         The proxy to use.
-     *
-     * @throws java.lang.UnsupportedOperationException
-     *         If this method is called after proxy settings have already been set or after at least 1 JDA object has been created.
-     *
-     * @return Returns the {@link net.dv8tion.jda.core.JDABuilder JDABuilder} instance. Useful for chaining.
-     */
-    public JDABuilder setProxy(HttpHost proxy)
+    // FIXME setHttpClientBuilder() & setWebsocketFactory() are ugly
+    // I don't like users having to mess with the ws or http lib,
+    // on the other side this allows them to configure the behaviour more deeply without
+    // making JDABuilder overly complicated by adding too many methods 
+    public JDABuilder setHttpClientBuilder(OkHttpClient.Builder httpClientBuilder)
     {
-        if (jdaCreated)
-            throw new UnsupportedOperationException("You cannot change the proxy after a JDA object has been created. Proxy settings are global among all instances!");
-        this.proxy = proxy;
+        this.httpClientBuilder = httpClientBuilder;
         return this;
     }
```

Comment:
I can't keep `setProxy` as the `HttpHost` class doesn't exist anymore. I laso don't really like the idea with the `WebSocketOptions` class anymore as it would just be a clone of the WebSocketFactory in the end (same for the http stuff).

@DV8FromTheWorld do you have any idea what we should do with this?